packages/components/src/styles/components/accordion.scss

```diff
@@ -36,8 +56,52 @@ $hds-accordion-item-border-radius: 6px;
     }
   }
 
-  &.hds-accordion-item--contains-interactive {
+  &.hds-accordion-item--contains-interactive,
+  &.hds-accordion-item--is-static {
     box-shadow: var(--token-surface-base-box-shadow);
+  }  
+}
+
+.hds-accordion-item--type-flush {
+  box-shadow: 0 1px 0 0 var(--token-color-border-primary);
+
+  &:first-child {
+    box-shadow: 0 1px 0 0 var(--token-color-border-primary), 0 -1px 0 0 var(--token-color-border-primary), ;
+  }
+
+  // we need this specificity to override the focus-ring mixin
+  .hds-accordion-item__button.hds-accordion-item__button--parent-does-not-contain-interactive::before {
+    border-radius: 0;
+  }
+
+  &.hds-accordion-item--does-not-contain-interactive:not(.hds-accordion-item--is-static)>.hds-disclosure-primitive__toggle {
+    &:hover,
+    &.mock-hover {
+      .hds-accordion-item__toggle {
+        background: var(--token-color-surface-interactive-hover);
+      }
+    }
+
+    &:active,
+    &.mock-active {
+      .hds-accordion-item__toggle {
+        background: var(--token-color-surface-interactive-active);
+      }
+    }
+  }
+}
+
+.hds-accordion-item__toggle-content {
+  .hds-accordion-item--is-static.hds-accordion-item--size-large & {
```

Comment:
One alternative could be to render anyway the button, but with a `disabled` attribute, and hide it (visually) in CSS when `[disabled]`(not sure if it would work from an accessibility standpoint). Just an idea.

Reply:
+1 That would remove the need for all these extra css styles and sass variables and tie the spacing to the actual button sizes.

Reply:
I generally try to avoid rendering elements in the DOM if they are not required (especially interactive ones), but I'll give this a go if you feel that it will simplify the code.

Reply:
> I generally try to avoid rendering elements in the DOM if they are not required (especially interactive ones)

I 100% agree, but one of the exceptions I tend to make is this use case, when the element leaves a (visual) hole in the UI, and workaround are needed to match the layout when the element is rendered: in that case, I tend to see if it's possible to just hide the element an let the layout engine to the work for me.

Reply:
Refactored in d12ac1f